pilot/pkg/xds/endpoint_builder.go

```diff
@@ -395,6 +390,17 @@ func buildEnvoyLbEndpoint(b *EndpointBuilder, e *model.IstioEndpoint) *endpoint.
 			},
 		},
 	}
+	if dir == model.TrafficDirectionInboundVIP {
+		// For inbound, we only use EDS for the VIP cases. The VIP cluster will point to encap listener.
+		//target := model.BuildSubsetKey(model.TrafficDirectionInboundPod, "", host.Name(e.Address), int(e.EndpointPort))
+		//addr = util.BuildInternalAddress(target)
+		address := e.Address
+		tunnelPort := 15008
+		// We will connect to inbound_CONNECT_originate internal listener, telling it to tunnel to ip:15008,
+		// and add some detunnel metadata that had the original port.
+		tunnelOrigLis := "inbound_CONNECT_originate"
```

Comment:
What about cases where they aren't tunneled? I guess this is making the assumption all pods behind a waypoint are tunneled, but I am not sure if that is a safe assumption to make

Reply:
Right, but the only way to have authz for backends is if all data paths to the backends go through the waypoint, and the waypoints require ztunnel.

Reply:
So the primary case I am thinking of is we have a Service spanning 2 clusters, 1 does not yet have ztunnel deployed (maybe not ever, although this is not a very good idea). 1 waypoint in 1 cluster

1. Current behavior: we always send to the waypoint (but its not enforced on one cluster!), then LB to any pod
2. This PR:   we always send to the waypoint (but its not enforced on one cluster!), then LB to any pod, cluster without ztunnel will fail
3. Small tweak to this PR: we always send to the waypont (but its not enforced on one cluster!), then LB *only to pods with a ztunnel*
4. Alternative: We LB between waypoint and direct to pods without ztunnel. Once they get a ztunnel, then traffic will also go through the waypoint.

I don't really like option (4), mostly for completeness. 2/3 may be the same if this PR already does that and I just missed it.

So it seems like in either case, you can bypass the waypoint? The choice seems to come down to whether or not we LB to all pods or only those with tunnel?

WDYT?

Reply:
LB shouldn't depend on policy, and the suggestion implies that LB will load balance differently if endpoint is firewalled with ztunnel or not. I think this is the same mistake as having authz after LB, no?

Reply:
I don't think option 2 is viable as it means when a waypoint is deployed some % of requests will fail. Only 1 and 3, to me, are acceptable

Reply:
Having a ztunnel is an authz policy, so it's expected that some requests fail. What are we trying to optimize here? Why would someone migrate a service partially to ztunnel?

Reply:
Option 2 doesn't actually enforce anything since you could send it directly to the pod anyways. So it seems like we do not get enforcement, but we do get failed requests, which seems undesired.

Its not that someone *should* have a service partially migrated for long term (although it will happen to ~100% of multicluster users for migration), but if they *do* we need to have reasonable behavior. That may be funky load balancing, but not requests dropped IMO

Reply:
There's some gap in the relation of a service and its endpoints I think. I'd imagine a waypoint would be a declaration that a service owns its endpoints (meaning ztunnel must be present on the server), and the sane thing to do for mis-configured backends is to reject requests to them (since otherwise, we might be violating an L4 policy).

Wayponts should not be bound to clusters at all IMHO, so they should be uniformly opted in or out across fleet.

Reply:
This PR is far from complete. I'm looking for gaps in my understanding and checking what fails.

Reply:
Is there an index to check if IP is behind ztunnel here?